### PR TITLE
[JBPM-9839] synchronize the contents (system properties) from standalone-*.xml with the default-kieserver.xml

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-9839

### CODE REVIEW

**Comment by elguardian** [comment]
Jenkins retest this please

**Comment by lazarotti** [comment]
Jenkins retest this please

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java [hunk 1]
```diff
@@ -196,6 +203,31 @@ protected void init(StartupStrategy startupStrategy) {
         eventSupport.fireAfterServerStarted(this);
     }
 
+    /**
+     * we allow some config items to be overwritten by properties
+     * @param currentState
+     */
+    private boolean updateMutableEnvironmentState(KieServerState currentState) {
+        boolean updated = false;
+        for(String key : Arrays.asList(KieServerConstants.CFG_KIE_CONTROLLER_TOKEN, KieServerConstants.CFG_KIE_CONTROLLER_PASSWORD, KieServerConstants.CFG_KIE_CONTROLLER_TOKEN)) {
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java:212]: Token is repeated and missing user:
```suggestion
        for(String key : Arrays.asList(KieServerConstants.CFG_KIE_CONTROLLER_USER, KieServerConstants.CFG_KIE_CONTROLLER_PASSWORD, KieServerConstants.CFG_KIE_CONTROLLER_TOKEN)) {
```

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java [hunk 2]
```diff
@@ -159,6 +160,12 @@ protected void init(StartupStrategy startupStrategy) {
 
         KieServerState currentState = repository.load(KieServerEnvironment.getServerId());
 
+        // we sync env with state just in case for some env entries only
+        // if there is an update we just need to store it again
+        if(updateMutableEnvironmentState(currentState)) {
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java:165]: Missing a blank 
```suggestion
        if (updateMutableEnvironmentState(currentState)) {
```

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java [hunk 3]
```diff
@@ -196,6 +203,31 @@ protected void init(StartupStrategy startupStrategy) {
...
+        boolean updated = false;
+        for(String key : Arrays.asList(KieServerConstants.CFG_KIE_CONTROLLER_TOKEN, KieServerConstants.CFG_KIE_CONTROLLER_PASSWORD, KieServerConstants.CFG_KIE_CONTROLLER_TOKEN)) {
+            updated |= updateConfigItemFromEnv(currentState, key);
+        }
+        return updated;
+    }
+
+    private boolean updateConfigItemFromEnv(KieServerState state, String key) {
+        String envValue = System.getProperty(key);
+        if(envValue == null) {
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java:220]: same, add blank

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java [hunk 4]
```diff
@@ -196,6 +203,31 @@ protected void init(StartupStrategy startupStrategy) {
...
+            return false;
+        }
+        if(state.getConfiguration().getConfigItem(key) != null && envValue.equals(state.getConfiguration().getConfigItem(key).getValue())) {
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java:223]: same, add blank

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java [hunk 5]
```diff
@@ -159,6 +160,12 @@ protected void init(StartupStrategy startupStrategy) {
 
         KieServerState currentState = repository.load(KieServerEnvironment.getServerId());
 
+        // we sync env with state just in case for some env entries only
+        // if there is an update we just need to store it again
+        if(updateMutableEnvironmentState(currentState)) {
+            repository.store(KieServerEnvironment.getServerId(), currentState);
```
Comment (gmunozfe) [inline kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java:166]: perhaps a log could be included here to indicate that the state has been updated due to changes in the controller access, wdyt?

**Review by gmunozfe — CHANGES_REQUESTED** [review]
Looks good, just a typo to fix because it's looping twice on token variable instead of user.
Could you add an integration test?

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java (hunk 2 above)
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java:165]: done

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java (hunk 5 above)
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java:166]: added to updateConfigItemFromEnv so we can have more fine detail there.

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java (hunk 1 above)
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java:212]: done

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java (hunk 3 above)
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java:220]: done

File: kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java (hunk 4 above)
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-common/src/main/java/org/kie/server/services/impl/KieServerImpl.java:223]: done

**Review by gmunozfe — APPROVED** [review]
Looks good to me, flawless work @elguardian !

**Pull request merged**